### PR TITLE
Des 413

### PR DESCRIPTION
**Pull request by StommePoes**
Code review. What I did:
1. Fixed unreported error on the demo page, "Open, then sort topics randomly for 5 seconds "
2. re-indented files to match current code standards
3. removed all the Collapse code and directories
4. added ContextualHelp.accordion() instead of Collapse to deal with accordions
5. adjusted README Accessibility section for accordions and general accessibility
6. added a manageLayers() function and a state-holder `_inner_trigger`. If a specific topic is shown after clicking on a help topic link, that is stored in `_innter_trigger` so focus can be returned to it. manageLayers() sets hidden layers to visibility: hidden which prevents AT and keyboard from accessing it while not needing the manual forced repaint of display: none.
7. openHelpTopics was modified to use manageLayers()
8. tried to unify the spacing in all the functions
9. running `npm test` fails now on various tests, I believe because I needed to use `setTimeout()` in manageLayers(). So I've added two of the failing tests to the demo/index.html page to prove they work, and as a model for the new Selenium tests we'll need from QA.

I think that's it. It has many elses; the Jeremy would be disappoint.

### CODE REVIEW

File: demo/src/demo.js [hunk 1]
```diff
@@ -1,88 +1,91 @@
...
+  }
+  var help = document.getElementById('o-contextual-help-drawer').oContextualHelp;
+  help.open();
+  function rndSort() {
+    var ts = help.getTopics(),
+        l = ts.length; 
+    ts.sort(function() {
+      return Math.floor(Math.random()*l) - 1;
+    });
+    help._el.querySelector('.o-contextual-help__excerpt-list').innerHTML = '';
```
Comment (StommePoes) [inline demo/src/demo.js]: This is the fix for the undocumented bug (openAndRandomSort only removes topics). removeAllTopics re-inits the con-help and this means we can never get more topics in again, no matter where we try to store them.

**Comment by wyseguyonline** [comment]
I don't see anything so it LGTM

**Comment by mhomolak** [comment]
Looks good!

**Comment by StommePoes** [comment]
@umahaea @udhadpa This is ready for QA.

There are 3 failing karma tests when `npm test` is run. Into `/demo/index.html` and /demo/src/demo.js` I have moved 2 of the failing tests: `window.directLoader` and `window.addRemoveTopics`. I believe they fail in karma because these both need to call my new `manageLayers()` function which has a setTimeout(), so these need to move to Selenium.

I thought the final failing test is only failing because the ones before it fail.

Additionally, there is an error after tests run (also seen in `master` branch, so this is not from my changes):
`TypeError: undefined is not an object (evaluating 'cData.title')`
This might be something to do with Phantom? Or it's because the final test passes no id and therefore the fetch (cData) comes back with nothing. However I have manually verified that cData.title exists, is correct and of the correct type manually when running the other tests.

In addition, we need some new tests:
1. That when the drawer is open, focus cycles through the visible layer (same as Drawer test)
2. That when a topic heading is clicked and the second layer opens, that focus is now on the "Back To Help Topics" button
3. That focus cycles now inside this new visible layer
4. That if #2 happened, then when "Back To Help Topics" is clicked, the topic heading in question has the focus
5. That if #2 did not happen, but the help was opened directly to a topic, that focus is on the "Back To Help Topics" button, and clicking it should see focus on the close button of the first layer (and help topics is displayed).

**Pull request merged**